src/main/scala/workflow/AutoCacheRule.scala

```diff
@@ -43,12 +43,17 @@ class AutoCacheRule(cachingMode: CachingStrategy) extends Rule with Logging {
    * Get an estimate for how many times each instruction will be executed, assuming
    * the given set of instructions have their outputs cached
    */
-  def getRuns(instructions: Seq[Instruction], cache: Set[Int], nodeWeights: Map[Int, Int]): Map[Int, Int] = {
-    val immediateChildrenByInstruction = getImmediateChildrenByInstruction(instructions)
```

Comment:
I narrowed it down using timings, and it was taking 4 seconds per time it was executed. `getRuns` ends up getting executed thousands of times.